monkey/common/cloud/aws/test_aws_instance.py

```diff
@@ -0,0 +1,260 @@
+
+# from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+INSTANCE_IDENTITY_DOCUMENT_RESPONSE = """
+{
+    "devpayProductCodes": null,
+    "marketplaceProductCodes": ["1abc2defghijklm3nopqrs4tu"],
+    "availabilityZone": "us-west-2b",
+    "privateIp": "10.158.112.84",
+    "version": "2017-09-30",
+    "instanceId": "i-1234567890abcdef0",
+    "billingProducts": null,
+    "instanceType": "t2.micro",
+    "accountId": "123456789012",
+    "imageId": "ami-5fb8c835",
+    "pendingTime": "2016-11-19T16:32:11Z",
+    "architecture": "x86_64",
+    "kernelId": null,
+    "ramdiskId": null,
+    "region": "us-west-2"
+}
+"""
+
+
+EXPECTED_INSTANCE_ID = 'i-1234567890abcdef0'
+
+EXPECTED_REGION = 'us-west-2'
+
+EXPECTED_ACCOUNT_ID = '123456789012'
+
+
+def get_test_aws_instance(text={'instance_id': None,
+                                'region': None,
+                                'account_id': None},
+                          exception={'instance_id': None,
+                                     'region': None,
+                                     'account_id': None}):
+    with requests_mock.Mocker() as m:
+        # request made to get instance_id
+        url = f'{AWS_LATEST_METADATA_URI_PREFIX}meta-data/instance-id'
+        m.get(url, text=text['instance_id']) if text['instance_id'] else m.get(
+            url, exc=exception['instance_id'])
+
+        # request made to get region
+        url = f'{AWS_LATEST_METADATA_URI_PREFIX}meta-data/placement/availability-zone'
+        m.get(url, text=text['region']) if text['region'] else m.get(
+            url, exc=exception['region'])
+
+        # request made to get account_id
+        url = f'{AWS_LATEST_METADATA_URI_PREFIX}dynamic/instance-identity/document'
+        m.get(url, text=text['account_id']) if text['account_id'] else m.get(
+            url, exc=exception['account_id'])
+
+        test_aws_instance_object = AwsInstance()
+        return test_aws_instance_object
+
+
+# all good data
+@pytest.fixture
+def good_data_mock_instance():
+    return get_test_aws_instance(text={'instance_id': INSTANCE_ID_RESPONSE,
+                                       'region': AVAILABILITY_ZONE_RESPONSE,
+                                       'account_id': INSTANCE_IDENTITY_DOCUMENT_RESPONSE})
+
+
+def test_is_instance_good_data(good_data_mock_instance):
+    assert good_data_mock_instance.is_instance()
+
+
+def test_get_cloud_provider_name_good_data(good_data_mock_instance):
+    assert good_data_mock_instance.get_cloud_provider_name() == Environment.AWS
+
+
+def test_get_instance_id_good_data(good_data_mock_instance):
+    assert good_data_mock_instance.get_instance_id() == EXPECTED_INSTANCE_ID
+
+
+def test_get_region_good_data(good_data_mock_instance):
+    assert good_data_mock_instance.get_region() == EXPECTED_REGION
+
+
+def test_get_account_id_good_data(good_data_mock_instance):
+    assert good_data_mock_instance.get_account_id() == EXPECTED_ACCOUNT_ID
```

Comment:
This can stay how it is or you could make it one test. It's only testing _one thing_, which is successful construction of an AWS instance. I think it's easier to read as one test, but there es a faction that believes in "one assert per test case".

Reply:
I agree but I'd rather leave it as it is here so it's consistent across the file.